client/src/main.jsx

```diff
@@ -130,37 +155,92 @@ const router = createBrowserRouter([
           }
         },
       },
+
       {
-        path: "/transactions_form",
-        element: <TransactionForm />,
+        path: "/budgets",
+        element: <Budgets />,
+        loader: async () => ({
+          budgets: await getBudgetsByUserId(),
+        }),
+      },
+      {
+        path: "/budgets/:id",
+        element: <BudgetDetails />,
+        loader: async ({ params }) => ({
+          budget: await getBudget(params.id),
+        }),
+      },
+      {
+        path: "/budgets_form",
+        element: <BudgetForm />,
         loader: async () => ({
           categories: await getCategoriesByUserId(),
         }),
         action: async ({ request }) => {
           const formData = await request.formData();
-          await addTransaction({
+          const formDataObject = {
             name: formData.get("name"),
-            date: formData.get("date"),
             amount: parseInt(formData.get("amount"), 10),
-            type: formData.get("type"),
+            startDate: formData.get("start_date"),
+            endDate: formData.get("end_date"),
             categoryId: parseInt(formData.get("category"), 10),
-          });
-          return redirect(`/transactions`);
+          };
+          const validatedData = budgetFormValidator(formDataObject);
+
+          if (Object.keys(validatedData).length > 0) {
+            return validatedData;
+          }
+
+          formDataObject.name = formDataObject.name.trim();
```

Comment:
**question**: Why are you calling trim after the validation and not before ?

Reply:
My bad, i put it before for editing budget but forgot for budget creation